tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -0,0 +1,4520 @@
+  miopenConvolutionMode_t c_mode = miopenConvolution;
+  int pad_h = 0, pad_w = 0, u = 0, v = 0, dilation_h = 0, dilation_w = 0;
+  wrap::miopenGetConvolutionDescriptor(conv_desc, &c_mode, &pad_h, &pad_w, &u,
+                                       &v, &dilation_h, &dilation_w);
+
+  uint64 hash_value = tensorflow::hash<int>()(c_mode);
+  hash_value =
+      tensorflow::Hash64Combine(hash_value, tensorflow::hash<int>()(pad_h));
+  hash_value =
+      tensorflow::Hash64Combine(hash_value, tensorflow::hash<int>()(pad_w));
+  hash_value =
+      tensorflow::Hash64Combine(hash_value, tensorflow::hash<int>()(u));
+  hash_value =
+      tensorflow::Hash64Combine(hash_value, tensorflow::hash<int>()(v));
+  hash_value = tensorflow::Hash64Combine(hash_value,
+                                         tensorflow::hash<int>()(dilation_h));
+  hash_value = tensorflow::Hash64Combine(hash_value,
+                                         tensorflow::hash<int>()(dilation_w));
+
+  return hash_value;
+}
+
+// Class to implement a cache of compiled fusion plans.
+class CachedFusionPlans {
+ public:
+  // Check if we already have a fusion_plan corresponding to the given hash
+  // value.
+  // If we do, then
+  //   return true (+ the cached fusion plan via given pointer)
+  // Else
+  //   create a new fusion plan descriptor,
+  //   associate it with the given hash value in the cache
+  //   return false (+ newly created fusion plan via given pointer)
+  static bool FindOrCreate(uint64 hash,
+                           miopenFusionPlanDescriptor_t* fusion_plan,
+                           miopenFusionDirection_t fusion_direction,
+                           miopenTensorDescriptor_t input_descriptor) {
+    mutex_lock lock{cached_plans_mutex};
+
+    bool found_cached_plan = false;
+
+    auto it = cached_plans.find(hash);
+    if (it != cached_plans.end()) {
+      *fusion_plan = it->second;
+      found_cached_plan = true;
+    } else {
+      auto status = wrap::miopenCreateFusionPlan(fusion_plan, fusion_direction,
+                                                 input_descriptor);
+      if (status != miopenStatusSuccess) {
+        LOG(FATAL) << "call to miopenCreateFusionPlan failed: "
+                   << ToString(status);
+      } else {
+        cached_plans[hash] = *fusion_plan;
+      }
+    }
+
+    return found_cached_plan;
+  }
+
+  // Need to figure out the right place to call this routine.
+  static void Clear() {
+    mutex_lock lock{cached_plans_mutex};
+
+    for (auto it : cached_plans) {
+      auto status = wrap::miopenDestroyFusionPlan(it.second);
+      if (status != miopenStatusSuccess) {
+        LOG(FATAL) << "call to miopenDestroyFusionPlan failed: "
+                   << ToString(status);
+      }
+    }
+
+    cached_plans.clear();
+
+    unsupported_plans.clear();
+  }
+
+  // Is the Fusion plan corresponding to this hash unsupported.
+  static bool isUnsupportedFusionPlan(uint64 hash) {
```

Comment:
With some rare exceptions (getters), functions should start with an upper case.

Reply:
fixed in the latest commit